### ISSUE TITLE
2D images misalignment

### ISSUE THREAD

**Issue by gonetz** [gonetz/GLideN64#800]
Large 2D images (e.g. backgrounds) rendered with TexRect command often have alignment issues between parts of the image. Example:
![gliden64_mariokart64](https://cloud.githubusercontent.com/assets/8983092/11272295/0530d376-8ef6-11e5-9501-1ec4bd23937b.jpg)

**Comment by AmbientMalice** [comment]
I'm assuming this somewhat more drastic alignment problem is related? https://github.com/gonetz/GLideN64/issues/471

![c679e270-f0f4-11e4-9499-d13667380d73](https://cloud.githubusercontent.com/assets/10704701/11272571/9c60bac2-8f19-11e5-9ea6-35852bd5b2e9.jpg)

**Comment by gonetz** [comment]
Fix in fix_2D feature branch:
https://github.com/gonetz/GLideN64/tree/fix_2D

Details:
http://gliden64.blogspot.ru/2015/11/improved-2d-and-other-fixes.html

Binaries to test:
https://drive.google.com/file/d/0B0YqMPjGo3B2aUpyekkxV0RjX3M/view?usp=sharing

**Comment by gonetz** [comment]
Yes Rocket is related, but not fixed by this fix. I'll check why.

**Comment by AmbientMalice** [comment]
>Yes Rocket is related, but not fixed by this fix. I'll check why.

On the bright side, it's not just Rocket affected by that particular misalignment "look", so if that bug is fixed, it'll identical squash texture bugs in a bunch of other games, too.

**Comment by gonetz** [comment]
I suppose that Rocket has a specific bug, otherwise it would be fixed as well.

**Comment by AmbientMalice** [comment]
Anyhow, @gonetz, it's a really good bit of coding you've done. It's a huge visual improvement.

**Comment by purplemarshmallow** [comment]
Conker's BFD does not like this feature. The minimap is broken. The circles are not round. During scenes with a lot of text performance drops below fullspeed. I'll do more testing.
![gliden64_conker_bfd_000](https://cloud.githubusercontent.com/assets/7283660/11274779/bdea1a60-8eda-11e5-80cd-6ae90743d7c2.jpg)
![gliden64_conker_bfd_001](https://cloud.githubusercontent.com/assets/7283660/11274781/c2eb1ed8-8eda-11e5-9664-329abb8f380c.jpg)

**Comment by gonetz** [comment]
I forgot to mention: Pokemon Stadium 2 is incompatible with that technique.

**Comment by gonetz** [comment]
2D in Rocket does not use bilinear filtering. The new drawing method applied only to filtered textures. Unfiltered textures must be aligned. So, this is really specific bug.

**Comment by purplemarshmallow** [comment]
In OOT some items are invisible. I put the lens of truth on the right c button but it's invisible.
![gliden64_the_legend_of_zelda_004](https://cloud.githubusercontent.com/assets/7283660/11276932/6fdb657c-8ee4-11e5-8a63-d5da5f1c49f0.jpg)

**Comment by purplemarshmallow** [comment]
Super smash bros many images are missing
![gliden64_smash_brothers_000](https://cloud.githubusercontent.com/assets/7283660/11276980/a7e2dfe0-8ee4-11e5-8ae4-133f3a4165ca.jpg)

**Comment by purplemarshmallow** [comment]
F-Zero X
![gliden64_f-zero_x_000](https://cloud.githubusercontent.com/assets/7283660/11277229/e5c0cbdc-8ee5-11e5-99e3-5e1a6dc151a2.jpg)

**Comment by gonetz** [comment]
Thanks! I'll look is it possible to fix.

**Comment by gonetz** [comment]
Please give me save for Super smash bros.

**Comment by purplemarshmallow** [comment]
SSB: https://drive.google.com/file/d/0B7Y6r4SpC_QQa09sdjYzRklXTmM/view?usp=sharing

Rayman 2 all 2D is missing
![gliden64_rayman_2_000](https://cloud.githubusercontent.com/assets/7283660/11294945/d243a9e4-8f67-11e5-9ac4-ab66591d035c.jpg)

**Comment by LuigiBlood** [comment]
Mario Artist Paint Studio still fails with transparency (also the stamps that clearly is still not working right as it replaces the overlays screen with white).
![gliden64__001](https://cloud.githubusercontent.com/assets/952602/11300606/c400271c-8f90-11e5-8517-66f66a9c781a.png)

**Comment by gonetz** [comment]
All known graphics issues fixed. Performance issues are possible, if you have weak hardware.
Thanks purplemarshmallow for testing.

Updated binaries:
https://drive.google.com/file/d/0B0YqMPjGo3B2WmtmalVFOEJCSGc/view?usp=sharing

**Comment by purplemarshmallow** [comment]
Glover has depth problems and black borders
savestate: https://drive.google.com/file/d/0B7Y6r4SpC_QQZTIwcllNMjJRQ0U/view?usp=sharing
![gliden64_glover_002](https://cloud.githubusercontent.com/assets/7283660/11301717/6248061c-8f99-11e5-8ae9-ed10c45325b4.jpg)

**Comment by purplemarshmallow** [comment]
Pokemon Stadium 2 boxes are black
![gliden64_pokemon_stadium_2_009](https://cloud.githubusercontent.com/assets/7283660/11301816/ff8d51d4-8f99-11e5-895c-168fe31adbd7.jpg)
![gliden64_pokemon_stadium_2_014](https://cloud.githubusercontent.com/assets/7283660/11301820/0401894c-8f9a-11e5-9777-ef8969fb504e.jpg)

**Comment by purplemarshmallow** [comment]
Top Gear Overdrive intro. At the snowblind studios logo it runs very slow I get 8 VI/s. The snow kills performance.

**Comment by gonetz** [comment]
Wow, you're fast in bugs hunting! :)

Yes, if frame uses lots of separate texrects, it will work slow.

I'll check other issues. Thanks!

**Comment by LuigiBlood** [comment]
Mario Artist Polygon Studio
Regular GLideN64 (2015/10/12) [I should mention that's the one I use regularly]:
![polymenu_1](https://cloud.githubusercontent.com/assets/952602/11307578/7681ad7a-8fb9-11e5-87da-83e4015f888d.png)
GLideN64 2Dfix2:
![polymenu_2](https://cloud.githubusercontent.com/assets/952602/11307586/84f76552-8fb9-11e5-8a60-3aea05fc8324.png)

There's also missing graphics in Talent Studio.

By the way should I make several issues for each Mario Artist games or should I do all-in-one? There are still bugs to begin with.

**Comment by gonetz** [comment]
> By the way should I make several issues for each Mario Artist games or should I do all-in-one? There are still bugs to begin with.

Please open a ticket for this regression bug. Other issues with this game can be described in another single ticket.

**Comment by standard-two-simplex** [comment]
Great bugfix!
Excitebike 64 was unaffected by the changes though.
![gliden64_excitebike64_000](https://cloud.githubusercontent.com/assets/12202580/11318767/25ed14e2-9060-11e5-9bb3-c9f163be9a9b.png)
There are a few more alignment issues in the intro and main menu.
Savestate in the main menu: https://drive.google.com/file/d/0B8aIJkrbGl-zVXZHN3NxZzhfaEU/view?usp=sharing

**Comment by gonetz** [comment]
Excitebike 64 needs this fix from master:
https://github.com/gonetz/GLideN64/commit/f7ee508c94fabd3485377c3bb0fd82758f37afba

I'll merge it to the next build.

**Comment by gonetz** [comment]
Fixed all reported issues.

New binary:
https://drive.google.com/file/d/0B0YqMPjGo3B2TG5GQkQtZWM3UE0/view?usp=sharing

**Comment by TheOlilink** [comment]
In Ocarina of Time, the HUD textures seems to render more blurry than usual. Probably rendering at a smaller size?

![](http://i.imgur.com/g2cx0Yc.png)

**Comment by gonetz** [comment]
Yes, small objects look worse than with usual method, not sure why.
They look worse with software render as well.
Comparison screen shot: GLideN64 with regular texrect drawing - new drawing method - software lle 
![hearts](https://cloud.githubusercontent.com/assets/8983092/11331311/240650b0-91e3-11e5-9f5b-ef33fa936a4c.png)

**Comment by standard-two-simplex** [comment]
Ogre battle backgrounds still render wrong.
![gliden64_ogrebattle64_000](https://cloud.githubusercontent.com/assets/12202580/11334363/bc68e264-91d4-11e5-9ebf-b95014bdd4f9.png)
I don't know if the sprites not matching might be related.
Savestate; https://drive.google.com/file/d/0B8aIJkrbGl-zNW9Jd0lYaHhseVk/view?usp=sharing

**Comment by gonetz** [comment]
Problem with sprites is not related to this feature. The feature fixes only texrects. Sprites are not texrects.

**Comment by standard-two-simplex** [comment]
![gliden64_marioparty3_000](https://cloud.githubusercontent.com/assets/12202580/11409083/dad28f1c-93bd-11e5-8a62-86acd3069469.png)
Mario Party 3 has been broken. Maybe this changes had something to do with it.

**Comment by gonetz** [comment]
It's very possible.

**Comment by gonetz** [comment]
Updated binary:
https://drive.google.com/file/d/0B0YqMPjGo3B2bUhkR1ZWYi0tYjA/view?usp=sharing

**Comment by theboy181** [comment]
What's updated?

**Comment by LuigiBlood** [comment]
Mario Artist Talent Studio has no menu visible in the talent mode (top mode in the main menu).

**Comment by standard-two-simplex** [comment]
MP3 issue fixed.

**Comment by AmbientMalice** [comment]
As a side effect, likely of https://github.com/gonetz/GLideN64/commit/1580aa040d6e389a3dc33dc84ce190d59a9ce903, the black lines in issues such as https://github.com/gonetz/GLideN64/issues/450 have been half-fixed. Not perfect, because they're sometimes visible, but the alignment is better.

![gliden64_tetrisphere_001](https://cloud.githubusercontent.com/assets/10704701/11464228/30a093cc-9778-11e5-9824-689a907fa6a8.jpg)

I did notice a problem, however. The text in 40 Winks, another game with black lines tends to be cut off at end right hand side, and I'm 99% sure it didn't do that before.

![gliden64_40_winks_001](https://cloud.githubusercontent.com/assets/10704701/11464246/7010cf04-9778-11e5-9a0c-2b0db38ea7ed.jpg)

**Comment by SwinglesMan** [comment]
Thanks for this. It even improves speed when using texture filtering options like smoothing 4 and x2. However, anything used from that section still cause lag spikes. Way less than before, but it still happens.

Also, 4K doesn't seem to give errors anymore :)

**Comment by oddMLan** [comment]
![oh](https://cloud.githubusercontent.com/assets/7278372/11516551/2a009d86-9841-11e5-9b87-fdfecd562ff0.png)
![oh2](https://cloud.githubusercontent.com/assets/7278372/11516552/2cdf50b0-9841-11e5-95fe-813a880ff290.png)

Conker's BFD is affected, too.

**Comment by SwinglesMan** [comment]
Killer Instinct Gold as well. VS screen, characters get divided into sections. And while fighting, Heads can get chopped off.

**Comment by frankyfife** [comment]
Super Mario 64 give me some strange font errors with the plugin provided up above (the pre-fix version too). PJ64 2.2.0.9999, RSP Interpreter by Iconoclast, Azimer XA2 WIP6, NRage 2.4.

![gliden64_super_mario_64_001](https://cloud.githubusercontent.com/assets/16177808/11614147/8d7f7eda-9c39-11e5-93a1-aa8f7ccb4f71.jpg)
![gliden64_super_mario_64_000](https://cloud.githubusercontent.com/assets/16177808/11614148/8d8072fe-9c39-11e5-8115-2d5367397d5d.jpg)

**Comment by theboy181** [comment]
I noticed this weird character bug on MK64 too. If you full screen and return it will display normally. Does it display properly if you switch to full screen and back?

**Comment by frankyfife** [comment]
Tried that, but nothing changed. With jabos direct3d 8 plugin everything is fine.

**Comment by gonetz** [comment]
@AmbientMalice @oddMLan  @frankyfife 
I can't reproduce the issues you reported. Do you use NVidia cards?

@SwinglesMan I can reproduce KI problem, thanks

**Comment by frankyfife** [comment]
@gonetz - yes, i use an gtx 980Ti nvidia card

**Comment by gonetz** [comment]
Anybody with ATI or Intel graphics - please test issues with missing last character in CBFD and 40 winks. SM64 issue also does not reproduced on my ATI card.

**Comment by weinerschnitzel** [comment]
SM64 pre-level text is invisible on Intel, but Conker's BFD text is readable

**Comment by baptiste0602** [comment]
I can't reproduce CBFD, 40 Winks or SM64 bug, everything is OK for me.
I have a Nvidia card (GTX670) with latest drivers (359.06).

**Comment by oddMLan** [comment]
I actually have an AMD Radeon R7 200, so I don't think this issue is vendor-exclusive.

**Comment by gonetz** [comment]
Ok, it's not vendor specific,  good.  Please reset settings to defaults,  set shades cache off and test again.

**Comment by weinerschnitzel** [comment]
Still get the SM64 issue with shader cache disabled. The (E) version has the messed up text, (U) has invisible text before levels.

**Comment by oddMLan** [comment]
@gonetz Disabling Antialasing fixes the missing last letter issue in Conker's BFD. Also 40 Winks as well. @AmbientMalice

**Comment by oddMLan** [comment]
2D Fix version:
![bilinear bad](https://cloud.githubusercontent.com/assets/7278372/11706373/6fa55b5e-9eb5-11e5-9826-cfb302380508.png)

GlideN64 rev.34fa719
![bilinear good](https://cloud.githubusercontent.com/assets/7278372/11706375/71a18d06-9eb5-11e5-8d0f-9bf4a41b8e03.png)

Is it just me or the 2D fix version is applying bilinear filtering to textures that shouldn't have it? I don't see the same problem in @frankyfife 's screenshot though.

I tested Super Mario 64 (U) version.

**Comment by gonetz** [comment]
> Disabling Antialasing fixes the missing last letter issue in Conker's BFD. Also 40 Winks as well.

Thanks!

>Is it just me or the 2D fix version is applying bilinear filtering to textures that shouldn't have it?

It should not be. 2D fix applied only for texrects with bilinear filter. It can be my bug of course, as with AA.

**Comment by gonetz** [comment]
Missing gfx with AA: found and fixed.

"applying bilinear filtering to textures that shouldn't have it" : can't reproduce

SM64 text issue: yes, it happens in LLE mode. The issue is not related to this feature, it presents in master also.

**Comment by weinerschnitzel** [comment]
@oddMLan are you referring to the black border that appears on the HUD objects? I suppose this is the same result that happens with the hearts in Zelda.

**Comment by oddMLan** [comment]
> @oddMLan are you referring to the black border that appears on the HUD objects?

Yes.

>These "black borders" show up in the U version, the E version's are left unfiltered.

Interesting, haven't tested the E version. Let me check on that.

**Comment by weinerschnitzel** [comment]
Maybe the texture cache with the black borders is leftover, and the the unfiltered textures are drawn on top? That would explain why a few black pixels would remain. Could you recreate a clean unfiltered and clean filtered screenshot for comparison with this anomaly?

**Comment by oddMLan** [comment]
I wanted to test Earthworm Jim 3D because I know this game's logo had an issue that this 2D Fix intended to address:

![m good 2d](https://cloud.githubusercontent.com/assets/7278372/11733284
Using GlideN64 rev.34fa719  you can clearly see the logo is broken up into tiles here that don't seem to align quite well.

But I tried it with 2D fix version.
/1fcb3e46-9f69-11e5-8fe6-3a06fd41441b.png)
![missing 2d](https://cloud.githubusercontent.com/assets/7278372/11733289/3037bd0e-9f69-11e5-88dc-ca05e6a4fc79.png)

There's no logo. Or any 2D elements at all.
![m menu](https://cloud.githubusercontent.com/assets/7278372/11733322/882428fe-9f69-11e5-9ae6-c25f571cb5f5.png)

This is supposed to be the game's file menu.

**Comment by weinerschnitzel** [comment]
Does turning AA off resolve the issues with Earthworm Jim? I think gonetz said he has a commit on the way

**Comment by oddMLan** [comment]
I don't have AA or Anisotropic filtering enabled.

also
![zzzzzzzz](https://cloud.githubusercontent.com/assets/7278372/11733383/04ecb978-9f6a-11e5-9034-e914a5878981.png)

I noticed both U and J versions dont filter the text in the file select menu so that won't serve as good comparison.

but I created a new comparison
2D Fix default settings + shader cache disabled
![bilinear bad](https://cloud.githubusercontent.com/assets/7278372/11733445/a6ae8c0a-9f6a-11e5-96d5-2283e32c6bfc.png)

2D Fix after running game with GlideN64 rev.34fa719 without resetting settings to defaults. Shader cache disabled. Before running game made sure there was no trace of htc cache or shader cache left.
![wrooong](https://cloud.githubusercontent.com/assets/7278372/11733452/acd177be-9f6a-11e5-9b89-19add5834d99.png)

**Comment by weinerschnitzel** [comment]
do these filtering issues have anything to do with how close the texture is to the front of the viewport? Perhaps SM64 (E) draws the HUD items closer to the screen than (E) and (J.) I do recall Rice Video having to translate the beating heart via hack to show up on screen. Would that explain why the last heart (that beats) is unfiltered why the others are filtered? Maybe a threshold can be modified or certain textures can be idenified by CRC to disable filtering? 

(Sorry if my jargon is off, I only know so much about computer gfx, and I am stabbing in the dark)

**Comment by purplemarshmallow** [comment]
Viewport problems only happen in HLE. In LLE the RSP does this work. If it's a viewport problem it should be correct in LLE.

**Comment by weinerschnitzel** [comment]
LLE or HLE does not make a difference in filtering the HUD objects in SM64. Guess that rules that one out.

**Comment by gonetz** [comment]
Yes, all texrects in Earthworm Jim 3D disappeared. Bad.

**Comment by gonetz** [comment]
New build:
https://drive.google.com/file/d/0B0YqMPjGo3B2NEU4cUgwZEdSWVE/view?usp=sharing

Fixed AA issues, Mario Tennis issue and Earthworm Jim 3D issue.

**Comment by RibShark** [comment]
On the topic of the SM64 HUD being filtered on NA and JP, this also happens on the Virtual Console releases (with the HUD being unfiltered on the EU version), so this is likely not an emulation error.

You can get the HUD to become unfiltered by changing Texture Enhancement to Store.

**Comment by oddMLan** [comment]
![fixed](https://cloud.githubusercontent.com/assets/7278372/11768227/50b99f9a-a183-11e5-977b-b3c2b04ca4ab.png)
Nice work! The logo is fixed and also not misaligned anymore.
But, I notice now there's some distortion around the end of the yellow letters that didn't happen without the 2D Fix. Is this some sort of side effect that can't be avoided?

>On the topic of the SM64 HUD being filtered on NA and JP, this also happens on the Virtual Console releases (with the HUD being unfiltered on the EU version), so this is likely not an emulation error.

![dac](https://cloud.githubusercontent.com/assets/7278372/11768226/4b27291c-a183-11e5-8cd3-7c51f88b7d79.png)

Well, Nintendo's emulators aren't known for being the most accurate, either.
But Angrylion's pixel-accurate plugin shows the HUD as unfiltered in the U version so

**Comment by gonetz** [comment]
Yes, text in Earthworm Jim looks worse than without 2D fix, can't say why.

**Comment by oddMLan** [comment]
Could you end up reproducing the filtered HUD problem in Super Mario 64 (U)?

**Comment by oddMLan** [comment]
![help](https://cloud.githubusercontent.com/assets/7278372/11768476/0d61cbde-a18b-11e5-8950-a9c99cf77a37.png)
Using the "store" option under texture enhancements seems to fix the text in EWJ3D

**Comment by AmbientMalice** [comment]
The 2D fix cleans up RE2 portraits, but causes a problem -- ugly lines around the edges. Like the texture is incorrectly wrapped.

![re2_n64](https://cloud.githubusercontent.com/assets/10704701/11773878/551c1790-a27b-11e5-84df-c2d52aaafcf6.png)

Wonder Project J2 has a problem. Note the window in the background. This is with Glide64 or GLideN64 with "store" textures enabled.

![glide64_wonder_project_j2_01](https://cloud.githubusercontent.com/assets/10704701/11773892/88ca1c0e-a27b-11e5-8c14-deec6307abf4.png)

Now this is the window with 2D fix. The overall image is improved, but the top right corner of the window is missing.

![gliden64_wonder_project_j2_000](https://cloud.githubusercontent.com/assets/10704701/11773910/acf48e5c-a27b-11e5-9740-4b41ffb3861e.jpg)

**Comment by gonetz** [comment]
>>Using the "store" option under texture enhancements seems to fix the text in EWJ3D, but then the seams in the logo become noticeable again, although not as much as without the 2D Fix version.

2D fix does not applied to enhanced textures. That is it just not used with "store" enabled .

Actually it's not quite correct, because "store" mode is an optimization, not enhancement.

**Comment by SwinglesMan** [comment]
-Turning on AA during games in 4K crashes the emu (only tried with Bomberman 64)

-Bomberman 64 2nd attack has 2D alignment issues

-Cruising World intro logo

-Dr Mario, don't know if it is related but lines are around character speech bubble on menu screen 
(could be a trans or uprez issue i dunno)

-Fighter's Destiny 2, effects in background effects in stages are segrated by lines

-Fighter's Destiny, little man top left has extra lines when you win a round and it shows replay. Background stage image doesn't blend gracefully.

-Flying Dragon, horizontal lines across screen

-KI Gold, better but lines still in characters on select screen

-Mace TDA, Verticle line down middle of screen in menu

-Mischief Makers, black lines around things

-Tetris Sphere

-Majora's mask, line in title screen logo

**Comment by gonetz** [comment]
> Very likely this commit is responsible. 1580aa0

Yes, it is this commit. The game renders this scene in "unfair" order, and one rect (with the green lines on the last screenshot) overwrites the other one with the part of the window. My algorithm can't handle overlapping rects. I see no good solution atm.

**Comment by AmbientMalice** [comment]
> The game renders this scene in "unfair" order, and one rect (with the green lines on the last screenshot) overwrites the other one with the part of the window. My algorithm can't handle overlapping rects. I see no good solution atm.

For what it's worth, it's a very small visual artifact on an otherwise MUCH better looking scene. For example, look at the right pole of the ladder. It's clipped before, but correctly rendered now. Such subtle errors are fixed all over the place, and accumulate to greatly improve the image quality.

**Comment by gonetz** [comment]
Most of issues with this feature are fixed. However, it has performance issues in some games. Thus I decided to make another experimental version, which should work faster. I've made two optimizations:
- currently each texrect is rendered right to aux buffer. It requires additional GL state changes. New version just stores rect info and renders all rects as a batch when whole image is loaded. Number of GL state changes reduced greatly.
- The optimization above will not work if image consist of only one rect. For this case I"ve made another optimization: single rect images will be rendered with traditional method, that is without rendering them to aux buffer with native resolution. It should reduce overheads when lots of separate texrect are used on scene.

The new build is there:
https://drive.google.com/file/d/0B0YqMPjGo3B2UUI1QjJBZUdqUWs/view?usp=sharing

Unfortunately, optimization for single rects is incompatible with some games, e.g with Pokemon Stadium. I planned to make that optimization optional, but since the feature currently has no handles for users, I've just build a version with only the first optimization:
https://drive.google.com/file/d/0B0YqMPjGo3B2dmlJcldhVEk5WUE/view?usp=sharing

Build GLideN64_2DFix6-2.7z should work as good as previous GLideN64_2DFix5.7z, only bit faster.
GLideN64_2DFix6-1.7z should be almost as fast as regular (master) version.

Delayed rendering is quite hard to implement and new bugs are possible. If there will be too issues, I'll return to previous version.

**Comment by AmbientMalice** [comment]
Both optimised builds suffer from HUD flickering in Turok 3 when looking at flickering light sources. (New game, look right. Pause and unpause to bring up HUD.)

**Comment by gonetz** [comment]
:(

**Comment by GyoMan** [comment]
Both optimized builds break Yoshi's Story...

**Comment by RibShark** [comment]
@gonetz FYI: The latest compiled versions of this branch are being flagged as a virus by Google Drive, so it is currently impossible to download them.

**Comment by gonetz** [comment]
Hmm, strange. Anyway, I need to solve reported issues or revert the optimizations.

**Comment by standard-two-simplex** [comment]
I noticed flickering in 2D in many games in this branch. I noticed flickering was reported before. Is it supposed to be fixed?

An easy way to reproduce it for me is to get into Excitebike 64s rider select menu, but I seen it in Smash Bros 64 too in-game (percentages and characters logo).

**Comment by standard-two-simplex** [comment]
After testing a little bit I noticed the flicker only happens with the mupen64plus version. Can anyone else reproduce?

**Comment by theboy181** [comment]
Loads of these images are fixed with the latest s2dex commits. :)

**Comment by theboy181** [comment]
Claire's Picture in resident evil looks ok in GLideN64 3.0 but not current 2dfix ???

**Comment by Jj0YzL5nvJ** [comment]
Top Gear Overdrive can be "fixed" with dithering. But is very tricky. #2451

https://imgur.com/5RaYWvg
https://imgur.com/ljguGDO
https://imgur.com/NS09McZ
https://imgur.com/4gT8WdF
